front/src/applications/operationalStudies/v2/components/ScenarioHeader.tsx

```diff
@@ -26,7 +33,28 @@ const ScenarioHeader = ({ scenarioName }: ScenarioHeaderProps) => {
     });
   };
 
-  console.log({ activeBoards });
+  const checkTruncation = useCallback(() => {
+    if (scenarioNameRef.current) {
+      setIsTruncated((prev) => ({
+        ...prev,
+        scenarioName: scenarioNameRef.current!.scrollWidth > scenarioNameRef.current!.clientWidth,
+      }));
+    }
+    if (usernameRef.current) {
+      setIsTruncated((prev) => ({
+        ...prev,
+        username: usernameRef.current!.scrollWidth > usernameRef.current!.clientWidth,
+      }));
+    }
```

Comment:
To avoid extra setState and re-renders
```suggestion
    setIsTruncated((prev) => ({
      scenarioName: scenarioNameRef.current
        ? scenarioNameRef.current.scrollWidth > scenarioNameRef.current.clientWidth
        : prev.scenarioName,
      username: usernameRef.current
        ? usernameRef.current.scrollWidth > usernameRef.current.clientWidth
        : prev.username,
    }));
```